api/app/stations.py

```diff
@@ -77,22 +84,61 @@ async def get_stations(
     return _get_stations_local()
 
 
+def set_weather_variables(parent, station_union):
+    """
+    TODO: add a comment! and explain this code - it's difficult to read!
```

Comment:
...?

Reply:
:) again - updated it before your comment. please have a look now.